bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/api/AgsStrategy.scala

```diff
@@ -80,34 +83,25 @@ object AgsStrategy {
    * results are valid along with all assignments of guide probes to stars.
    */
   case class Selection(posAngle: Angle, assignments: List[Assignment]) {
-
     /**
      * Creates a new TargetEnvironment with guide stars for each assignment in
      * the Selection.
      */
     def applyTo(env: TargetEnvironment): TargetEnvironment = {
-      def findMatching(gpt: GuideProbeTargets, target: SPTarget): Option[SPTarget] = {
-        def name(t: SPTarget): Option[String] =
-          Option(t.getTarget.getName).map(_.trim)
-
-        name(target).flatMap(n => gpt.getTargets.asScalaList.find(name(_).exists(_ == n)))
-      }
-
-      (env /: assignments) { (curEnv, ass) =>
-        val target = new SPTarget(HmsDegTarget.fromSkyObject(ass.guideStar.toOldModel))
-        val oldGpt = curEnv.getPrimaryGuideProbeTargets(ass.guideProbe).asScalaOpt
-
-        val newGpt = oldGpt.fold(GuideProbeTargets.create(ass.guideProbe, target)) { gpt =>
-          // We already have guide probe targets for guide probe.
-          // Does one with the same target name already exist? If so, mark it as
-          // primary and replace it with the target we just made.  If not, add the
-          // target and mark it as primary.
-          findMatching(gpt, target).fold(gpt.update(OptionsList.UpdateOps.appendAsPrimary(target))) { existing =>
-            gpt.selectPrimary(existing).setPrimary(target)
-          }
-        }
-
-        curEnv.putPrimaryGuideProbeTargets(newGpt)
+      val targetMap = ==>>.fromList(assignments.map { case Assignment(gp,gs) =>
+        gp -> new SPTarget(HmsDegTarget.fromSkyObject(gs.toOldModel))})
+      val newAuto: AutomaticGroup = AutomaticGroup.Active(targetMap)
+
+      // If this is different from the old automatic GG, then replace.
+      val oldGuideEnvironment = env.getGuideEnvironment
+      val oldGuideEnv         = oldGuideEnvironment.guideEnv
+
+      if (oldGuideEnv.auto =/= newAuto) {
```

Comment:
That's a good point.  You can't edit auto targets via the UI.  It doesn't *feel* that satisfying though since you can certainly edit them programatically.  Still, checking the name might be the best option for now.  Once the `SPTarget` holds a mutable reference to an immutable `Target`, then we could just directly compare the `Target` references.  Alternatively today we'd need to do something like we do in the `AlmostEquals` implementation for `SPTarget` and compare them field-by-field.  I guess let's just go with the name check.